sqlx_test.go

```diff
@@ -1814,3 +1814,33 @@ func BenchmarkRebindBuffer(b *testing.B) {
 		rebindBuff(DOLLAR, q2)
 	}
 }
+
+func TestSelectReset(t *testing.T) {
+	RunWithSchema(defaultSchema, t, func(db *DB, t *testing.T) {
+		loadDefaultFixture(db, t)
+
+		filledDest := []string{"a", "b", "c"}
+		err := db.Select(&filledDest, "SELECT first_name FROM person ORDER BY first_name ASC;")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(filledDest) != 2 {
+			t.Errorf("Expected 2 first names, got %d.", len(filledDest))
+		}
+		expected := []string{"Jason", "John"}
+		for i, got := range filledDest {
+			if got != expected[i] {
+				t.Errorf("Expected %d result to be %s, but got %s.", i, expected[i], got)
+			}
+		}
+
+		var emptyDest []string
+		err = db.Select(&emptyDest, "SELECT first_name FROM person WHERE first_name = 'Jack';")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(emptyDest) != 0 {
```

Comment:
Check that this is nil instead to lock in the behavior that selecting 0 rows into a nil target doesn't need to create a non-nil slice.